stablehlo/dialect/StablehloBytecode.cpp

```diff
@@ -25,6 +25,28 @@ limitations under the License.
 // Debug Trace Helpers
 //===----------------------------------------------------------------------===//
 
+// Enable logging with flag:
+//   stablehlo-opt -debug-only=stablehlo-bytecode [...]
+//
+// Extract after function name, remove namespace.
+//   Called: write(mlir::stablehlo::TokenType, mlir::DialectBytecodeWriter ...
+//   ***Not Implemened: write(...
+#define _EXTRACT_AFTER(a, b) \
+  llvm::StringRef(a).substr(llvm::StringRef(a).find(b))
+
+#define _LOG_CALL_TO(func)                                                    \
+  DEBUG_WITH_TYPE(                                                            \
+      "stablehlo-bytecode",                                                   \
+      llvm::errs() << "Called: " << _EXTRACT_AFTER(__PRETTY_FUNCTION__, func) \
+                   << '\n')
+
+#define LOG_WRITE_CALL _LOG_CALL_TO("write")
+#define LOG_READ_CALL _LOG_CALL_TO(__func__)
+#define LOG_NOT_IMPLEMENTED \
+  DEBUG_WITH_TYPE(          \
+      "stablehlo-bytecode", \
+      llvm::errs() << "***Not Implemented: " << __PRETTY_FUNCTION__ << '\n')
+
```

Comment:
Hmm you might've forgotten to push the fix that removes this diff?

Reply:
Oh geez the whole thing is duplicated. I thought just the comment header was duplicated. Fixed it this time.